### ISSUE TITLE
`'!_debugLocked': is not true.` exception when using `PopScope`

### ISSUE THREAD

**Issue by Mamena2020** [flutter/flutter#139013]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

I'm trying to close the alert dialog and execute some function, but seems working fine in flutter `3.13.6`, but `NOT` working in Flutter `3.16.0` .  
 
I've tried switching between execution functions before exiting the alert dialog, but it still doesn't work.

```` dart
   /// exist alert dialog
  Future<void> exitAlertDialog({required BuildContext context}) async {
    await WidgetUtils.alertDialog3(
        context: context,
        title: "Save",
        body: const Text(
          "Save before exit?",
          textAlign: TextAlign.center,
        ),
        cancelText: "Cancel",
        primaryText: "Save",
        secondaryText: "Leave",
        isSecondaryDanger: true,
        onPrimary: () async {
          if (project.isPlaying) {
            await stopPlaying();
          }
          if (project.isRecording) {
            await stopRecord();
          }
          // save and exit
          // ignore: use_build_context_synchronously
          await save(
              context: context,
              onDone: () {
                Navigator.of(context).pop();
              });
          // ignore: use_build_context_synchronously
        },
        onSecondary: () async {
          if (project.isPlaying) {
            await stopPlaying();
          }
          if (project.isRecording) {
            await stopRecord();
          }
          //close and exit;
          // ignore: use_build_context_synchronously
          Navigator.of(context).pop();
        });
  }
````

### Expected results

Should be fine, when exit and execute function in alert dialog.

### Code sample

```dart
  class WidgetUtils{
  /// display alert dialog with 3 option button
  static Future<void> alertDialog3(
      {required BuildContext context,
      required String title,
      required Widget body,
      Alignment titleAlignment = Alignment.center,
      String cancelText = "cancel",
      String primaryText = "confirm 1",
      String secondaryText = "confirm 2",
      bool disposeOnCancel = true,
      bool disposeOnPrimary = true,
      bool disposeOnSecondary = true,
      Function? onPrimary,
      Function? onSecondary,
      Function? onCancel,
      bool showPrimaryButton = true,
      bool showSecondaryButton = true,
      bool isPrimaryDanger = false,
      bool isSecondaryDanger = false}) {
    return showDialog(
      context: context,
      useSafeArea: true,
      barrierDismissible: false,
      builder: (BuildContext dialogContext) {
        return AlertDialog(
          scrollable: true,
          title: Align(alignment: titleAlignment, child: Text(title)),
          content: body,
          actions: [
            TextButton(
              child: Text(
                cancelText,
                style: const TextStyle(color: Colors.grey),
              ),
              onPressed: () async {
                if (onCancel != null) {
                  onCancel();
                }
                if (disposeOnCancel) {
                  // ignore: use_build_context_synchronously
                  Navigator.pop(dialogContext);
                }
              },
            ),
            if (showSecondaryButton)
              TextButton(
                child: Text(
                  secondaryText,
                  style: TextStyle(
                      color: isSecondaryDanger
                          ? Colors.red.shade800
                          : Colors.grey.shade800),
                ),
                onPressed: () async {
                  if (disposeOnSecondary) {
                    // ignore: use_build_context_synchronously
                    Navigator.pop(dialogContext);
                  }
                  if (onSecondary != null) {
                    onSecondary();
                  }
                },
              ),
            if (showPrimaryButton)
              TextButton(
                child: Text(
                  primaryText,
                  style: TextStyle(
                      color:
                          isPrimaryDanger ? Colors.red.shade800 : Colors.black),
                ),
                onPressed: () {
                  if (disposeOnPrimary) {
                    // ignore: use_build_context_synchronously
                    Navigator.pop(dialogContext);
                  }
                  if (onPrimary != null) {
                      onPrimary();
                  }
                },
              ),
          ],
        );
      },
    );
  }
}
```

</details>

### Logs

```console
E/flutter (19598): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: 'package:flutter/src/widgets/navigator.dart': Failed assertion: line 4782 pos 12: '!_debugLocked': is not true.
E/flutter (19598): #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
E/flutter (19598): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
E/flutter (19598): #2      NavigatorState._pushEntry (package:flutter/src/widgets/navigator.dart:4782:12)
E/flutter (19598): #3      NavigatorState.push (package:flutter/src/widgets/navigator.dart:4739:5)
E/flutter (19598): #4      showDialog (package:flutter/src/material/dialog.dart:1423:65)
E/flutter (19598): #5      WidgetUtils.alertDialog3 (package:upline/utils/widget_utils.dart:114:12)
E/flutter (19598): #6      _ProjectManageViewState.exitAlertDialog (package:upline/views/studio/studio_view.dart:856:23)
E/flutter (19598): #7      _ProjectManageViewState.build.<anonymous closure> (package:upline/views/studio/studio_view.dart:905:9)
E/flutter (19598): #8      ModalRoute.onPopInvoked (package:flutter/src/widgets/routes.dart:1574:30)
E/flutter (19598): #9      _RouteEntry.pop (package:flutter/src/widgets/navigator.dart:3083:11)
E/flutter (19598): #10     NavigatorState.pop (package:flutter/src/widgets/navigator.dart:5251:13)
E/flutter (19598): #11     _ProjectManageViewState.exitAlertDialog.<anonymous closure> (package:upline/views/studio/studio_view.dart:893:33)
E/flutter (19598): #12     WidgetUtils.alertDialog3.<anonymous closure>.<anonymous closure> (package:upline/utils/widget_utils.dart:155:32)
E/flutter (19598): <asynchronous suspension>
```

</details>

### Flutter Doctor output

Doctor output

```console
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.16.0, on Microsoft Windows [Version 10.0.19045.3693], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.2.2)
[√] Android Studio (version 2021.2)
[√] VS Code (version 1.84.2)
[√] Connected device (4 available)
[√] Network resources

• No issues found!
```

</details>

**Comment by iapicca** [comment]
@Mamena2020 
this shouldn't be ignored

```dart
                  // ignore: use_build_context_synchronously
                  Navigator.pop(dialogContext);
```

the error is likely caused by the implementation and not by flutter,
can you provide a runnable code sample (not just a snippet of your code)?

**Comment by Mamena2020** [comment]
@iapicca  However, the error occurred after I upgraded Flutter from version `3.13.6` to `3.16.0`. Consequently, I find myself in a state of confusion as to which earlier version worked without issues, while the latest version seems to trigger errors.

My objective is to navigate back after closing an alert dialog. However, when using `Navigator.of(context).pop();` within the `onSecondary` callback, it results in an error.

``` dart
    await WidgetUtils.alertDialog3(
        context: context,
        title: "Save",
        body: const Text(
          "Save before exit?",
          textAlign: TextAlign.center,
        ),
        cancelText: "Cancel",
        primaryText: "Save",
        secondaryText: "Leave",
        isSecondaryDanger: true,
        onPrimary: () async {
          // save and exit
          await save(
              context: context,
              onDone: () {
                Navigator.of(context).pop();
              });
        },
        onSecondary: () async {
          //close and exit;
          // ignore: use_build_context_synchronously
          Navigator.of(context).pop(); // cousing error in latest version <<<<<====
        });
```

**Comment by huycozy** [comment]
@Mamena2020 I assemble your snippet code into this complete one. But I can't reproduce the issue on the latest Flutter channels (stable/master). 

Steps I did: 
1. Open dialog
2. Click on `Leave` button which triggers `onSecondary`

Actual result: The dialog is dismissed without error.


<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';

void main() async {
  runApp(
    MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyApp(),
    ),
  );
}

class MyApp extends StatefulWidget {
  const MyApp({
    super.key,
  });

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: TextButton(
          onPressed: () async {
            await WidgetUtils.alertDialog3(
                context: context,
                title: "Save",
                body: const Text(
                  "Save before exit?",
                  textAlign: TextAlign.center,
                ),
                cancelText: "Cancel",
                primaryText: "Save",
                secondaryText: "Leave",
                isSecondaryDanger: true,
                onPrimary: () async {
                  // save and exit
                  // await save(
                  //     context: context,
                  //     onDone: () {
                  //       Navigator.of(context).pop();
                  //     });
                },
                onSecondary: () async {
                  //close and exit;
                  Navigator.of(context).pop(); // cousing error in latest version <<<<<====
                });
          },
          child: Text('Hello'),
        ),
      ),
    );
  }
}

class WidgetUtils {
  /// display alert dialog with 3 option button
  static Future<void> alertDialog3(
      {required BuildContext context,
      required String title,
      required Widget body,
      Alignment titleAlignment = Alignment.center,
      String cancelText = "cancel",
      String primaryText = "confirm 1",
      String secondaryText = "confirm 2",
      bool disposeOnCancel = true,
      bool disposeOnPrimary = true,
      bool disposeOnSecondary = true,
      Function? onPrimary,
      Function? onSecondary,
      Function? onCancel,
      bool showPrimaryButton = true,
      bool showSecondaryButton = true,
      bool isPrimaryDanger = false,
      bool isSecondaryDanger = false}) {
    return showDialog(
      context: context,
      useSafeArea: true,
      barrierDismissible: false,
      builder: (BuildContext dialogContext) {
        return AlertDialog(
          scrollable: true,
          title: Align(alignment: titleAlignment, child: Text(title)),
          content: body,
          actions: [
            TextButton(
              child: Text(
                cancelText,
                style: const TextStyle(color: Colors.grey),
              ),
              onPressed: () async {
                if (onCancel != null) {
                  onCancel();
                }
                if (disposeOnCancel) {
                  Navigator.pop(dialogContext);
                }
              },
            ),
            if (showSecondaryButton)
              TextButton(
                child: Text(
                  secondaryText,
                  style: TextStyle(color: isSecondaryDanger ? Colors.red.shade800 : Colors.grey.shade800),
                ),
                onPressed: () async {
                  if (disposeOnSecondary) {
                    Navigator.pop(dialogContext);
                  }
                  if (onSecondary != null) {
                    onSecondary();
                  }
                },
              ),
            if (showPrimaryButton)
              TextButton(
                child: Text(
                  primaryText,
                  style: TextStyle(color: isPrimaryDanger ? Colors.red.shade800 : Colors.black),
                ),
                onPressed: () {
                  if (disposeOnPrimary) {
                    Navigator.pop(dialogContext);
                  }
                  if (onPrimary != null) {
                    onPrimary();
                  }
                },
              ),
          ],
        );
      },
    );
  }
}

```
</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.16.0, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.16.0 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db7ef5bf9f (8 hours ago), 2023-11-15 11:25:44 -0800
    • Engine revision 74d16627b9
    • Dart version 3.2.0
    • DevTools version 2.28.2

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.8+0-17.0.8b1000.22-10799086)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.76.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 119.0.6045.123

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

```bash
[!] Flutter (Channel master, 3.17.0-15.0.pre.28, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.17.0-15.0.pre.28 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2949638e82 (2 hours ago), 2023-11-26 19:35:25 -0500
    • Engine revision ebebb25aaa
    • Dart version 3.3.0 (build 3.3.0-152.0.dev)
    • DevTools version 2.30.0-dev.4
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.8+0-17.0.8b1000.22-10799086)

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe Patch 3.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.76.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 119.0.6045.159
    ! Error: iPhone has recently restarted. Xcode will continue when iPhone is unlocked. (code -14)

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```
</details>

**Comment by Mamena2020** [comment]
@huycozy @iapicca Upon further debugging, I discovered that an `error` occurred when I utilized `PopScope` in the latest version, `3.16.0`, as a replacement for `WillPopScope` in the older version. Interestingly, removing `PopScope` resulted in the smooth operation of the application. Is there any solution for this issue?

code using PopScope

``` dart

    @override
  Widget build(BuildContext context) {
    return PopScope(
        onPopInvoked: (_) {
        await WidgetUtils.alertDialog3(
                context: context,
                title: "Save",
                body: const Text(
                  "Save before exit?",
                  textAlign: TextAlign.center,
                ),
                cancelText: "Cancel",
                primaryText: "Save",
                secondaryText: "Leave",
                isSecondaryDanger: true,
                onPrimary: () async {
                },
                onSecondary: () async {
                  //close and exit;
                  Navigator.of(context).pop(); // cousing error in latest version <<<<<====
                });
        },
        canPop: false,
        child: Scaffold(
```
flutter doctor -v
``` console

  [√] Flutter (Channel stable, 3.16.0, on Microsoft Windows [Version 10.0.19045.3693], locale en-US)
  • Flutter version 3.16.0 on channel stable at E:\PROGRAMING\FLUTTER\flutter-sdk
  • Upstream repository https://github.com/flutter/flutter.git
  • Framework revision db7ef5bf9f (12 days ago), 2023-11-15 11:25:44 -0800
  • Engine revision 74d16627b9
  • Dart version 3.2.0
  • DevTools version 2.28.2

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\android-sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = C:\android-sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.2.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32519.379
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] VS Code (version 1.84.2)
    • VS Code at C:\Users\admin\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.76.0

[√] Connected device (4 available)
    • Redmi Note 6 Pro (mobile) • 192.168.1.10:5555 • android-arm64  • Android 9 (API 28)
    • Windows (desktop)         • windows           • windows-x64    • Microsoft Windows [Version 10.0.19045.3693]
    • Chrome (web)              • chrome            • web-javascript • Google Chrome 119.0.6045.160
    • Edge (web)                • edge              • web-javascript • Microsoft Edge 119.0.2151.72

[√] Network resources
    • All expected network resources are available.

• No issues found!
```

**Comment by goderbauer** [comment]
cc @justinmc @chunhtai

**Comment by KuroodoD** [comment]
I also had the error when trying to show a dialogue via `onPopInvoked` with PopScope. I got past the error by showing the dialogue as a post frame callback.

```
onPopInvoked: (_) {
        WidgetsBinding.instance.addPostFrameCallback(
          (_) async {
            // show your dialog here
          },
        );
      },
```

**Comment by ftprashant** [comment]
I had an issue when the screen returned with value from the current screen.

 Navigator.pop(context,value);

**Comment by Mamena2020** [comment]
this issues has been fix in flutter `3.16.2`
``` console
[√] Flutter (Channel stable, 3.16.2, on Microsoft Windows [Version 10.0.19045.3693], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.2.2)
[√] Android Studio (version 2021.2)
[√] VS Code (version 1.85.0)
[√] Connected device (3 available)
[√] Network resources

• No issues found!
```

**Comment by AndersonMichellin** [comment]
> I also had the error when trying to show a dialogue via `onPopInvoked` with PopScope. I got past the error by showing the dialogue as a post frame callback.
> 
> ```
> onPopInvoked: (_) {
>         WidgetsBinding.instance.addPostFrameCallback(
>           (_) async {
>             // show your dialog here
>           },
>         );
>       },
> ```

This solution is good, but it didn´t resolve my problem. Because sometimes my project closes all pages and I exit the application.
The PopScope behavior with WidgetsBinding is different as WillPopScope.
We need to solve this problem, of course that is only a warning message.

**Comment by ftprashant** [comment]
> this issues has been fix in flutter `3.16.2`
> 
> ```
> [√] Flutter (Channel stable, 3.16.2, on Microsoft Windows [Version 10.0.19045.3693], locale en-US)
> [√] Windows Version (Installed version of Windows is version 10 or higher)
> [√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
> [√] Chrome - develop for the web
> [√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.2.2)
> [√] Android Studio (version 2021.2)
> [√] VS Code (version 1.85.0)
> [√] Connected device (3 available)
> [√] Network resources
> 
> • No issues found!
> ```

we already use this version still replicate issue

**Comment by withmukeshjoshi** [comment]
Facing same issue in 3.16.4 as well.


E/flutter ( 3563): #20     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:141:18)
E/flutter ( 3563): #21     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:127:7)
E/flutter ( 3563): #22     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:488:19)
E/flutter ( 3563): #23     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:468:22)
E/flutter ( 3563): #24     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:439:11)
E/flutter ( 3563): #25     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:413:7)
E/flutter ( 3563): #26     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:376:5)
E/flutter ( 3563): #27     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:323:7)
E/flutter ( 3563): #28     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:292:9)
E/flutter ( 3563): #29     _invoke1 (dart:ui/hooks.dart:328:13)
E/flutter ( 3563): #30     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:410:7)
E/flutter ( 3563): #31     _dispatchPointerDataPacket (dart:ui/hooks.dart:262:31)
